gc.c

```diff
@@ -11822,7 +11826,10 @@ ruby_gc_set_params(void)
 
     /* RUBY_GC_HEAP_INIT_SLOTS */
     if (get_envparam_size("RUBY_GC_HEAP_INIT_SLOTS", &gc_params.heap_init_slots, 0)) {
-        gc_set_initial_pages(objspace);
+        gc_set_initial_pages(objspace, TRUE);
+    }
+    else {
+        gc_set_initial_pages(objspace, FALSE);
```

Comment:
Why is this needed? The setup should have been done in `Init_heap` so we don't need to do it again.

Reply:
It's so that if you don't set `RUBY_GC_HEAP_INIT_SLOTS` but do set any `RUBY_GC_HEAP_INIT_SIZE_%d_SLOTS` we'll set initial pages.